uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-action-dropdown.spec.js

```diff
@@ -0,0 +1,1541 @@
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+
+          describe('when expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, true);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+        });
+
+        describe('when status is "rejected"', function () {
+          beforeEach(function () {
+            setRequestStatus(request, 'rejected');
+          });
+
+          describe('when not expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, false);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+
+          describe('when expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, true);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+        });
+
+        describe('when status is "cancelled"', function () {
+          beforeEach(function () {
+            setRequestStatus(request, 'cancelled');
+          });
+
+          describe('when not expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, false);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+
+          describe('when expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, true);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+        });
+      });
+    });
+
+    describe('wants to change status of leave request', function () {
+      // Any action, role or request could be specified here
+      var action = 'approve';
+
+      beforeEach(function () {
+        // Any role or request could be specified here
+        role = 'admin';
+        request = getRequestByType('leave');
+
+        spyOn(request, action).and.returnValue($q.resolve());
+        resolveDialogWith(null);
+        vm.act(action);
+        $rootScope.$digest();
+        compileComponent();
+      });
+
+      it('sees a confirmation dialog', function () {
+        expect(dialog.open).toHaveBeenCalled();
+      });
+
+      describe('confirms the action', function () {
```

Comment:
Please also test the case when the user dismisses the dialog, to make sure that indeed nothing happens